### ISSUE TITLE
[Feature Request] Support for finishing the compile with partial errors

### ISSUE THREAD

**Issue by AlirezaTheH** [xu-cheng/latex-action#73]
Sometimes latex generates a 'Illegal unit of measure' error when there is no error present. This is because it does not recognise some common lengths like `\linewidth` as containg a unit. The pdf result is still fine if we don't set `-halt-on-error` argument. I think it would be nice to have a new argument like `ignore_errors` to resolve this issue.
I also would be happy to submit a PR to add this feature.

Also I wanted to thank you for this great acction @xu-cheng.

**Comment by xu-cheng** [comment]
> The pdf result is still fine if we don't set -halt-on-error argument.

You can custom `args` input to control what flags are passed to `latexmk`. 

It is not clear to me what `ignore_errors` would behave. Will it control whether to continue to run the rest of latexmk execution? Or will it be mean to build next document for when building multiple documents? 

For the former, we may introduce a `latexmk_not_halt_on_error` argument. Can you confirm your issue would be resolved if `-halt-on-error` is not passed to `latexmk`?

For the latter, we can add a `continue_on_error` argument. With it is set, the action would continue building other documents but still exits with error code if there is any error.

**Comment by AlirezaTheH** [comment]
> It is not clear to me what `ignore_errors` would behave. Will it control whether to continue to run the rest of latexmk execution? Or will it be mean to build next document when building multiple documents?

It's the former.
 
> For the former, we may introduce a `latexmk_not_halt_on_error` argument. Can you confirm your issue would be resolved if `-halt-on-error` is not passed to `latexmk`?

Yes, if `-halt-on-error` is not passed latexmk, still gives return code `1` which causes the action to be failed.

**Comment by xu-cheng** [comment]
>  the problem is the return code 1 .

I think the return code should be 1 in this case as these are real errors and LaTeX engine does return error code. 

From what you are telling me so far, I felt that the real issue is that there are some incorrect usages in your latex code. The "Illegal unit of measure" message should not be ignored.

**Comment by AlirezaTheH** [comment]
No, as I mentioned before, Sometimes latex generates a 'Illegal unit of measure' error when there is no error present. This is because it does not recognise some common lengths like `\linewidth` as containg a unit.
See [here](https://github.com/liantze/AltaCV/blob/23a4dff8b8fcfe899264b5b7620b20613cc16056/altacv.cls#L358) for example. Also forcing a unit in this occasion will make issues in resulted PDF, so It's not an option.

**Comment by fs-c** [comment]
Another perspective on this feature request: My use case is building hundreds of (unmaintained) .tex files for archival purposes, some of which contain errors. I don't particularly care if one or two files contain an error, I'll just link to the .tex file in that case or something. It looks like this request was abandoned over LaTeX semantics, so I am mentioning this to show that sometimes people really don't care about errors. (At least not enough to stop the other 300 files from being built.)

I hope this is the most recent issue on the topic, I haven't found any others.

**Comment by xu-cheng** [comment]
@fs-c For your usecase, I think a `continue_on_error` option is warranted when building multiple documents. Would you mind to create a PR?

**Comment by fs-c** [comment]
@xu-cheng Thanks for the quick reply, will do.
